### PR TITLE
code refactor and bug fixes

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary: 
- changed name of folder from inventory to inventories generated on linchpin init
- dropped linchpin_config.yml on linchpin init
- removed topology_upstream , layout_upstream attributes in PinFile . However, user can specify either using --upstream while using linchpin topology get or linchpin layout get.
- changed variable names and references of lpf to pf ( pinfile ) for consistency
- updated requirements.txt

### CODE REVIEW

**Comment by samvarankashyap** [comment]
centos-paas-test

**Comment by samvarankashyap** [comment]
centos-paas-test

**Comment by samvarankashyap** [comment]
centos-paas-test

**Comment by samvarankashyap** [comment]
centos-paas-test

**Comment by samvarankashyap** [comment]
centos-paas-test

**Comment by samvarankashyap** [comment]
centos-paas-test

**Comment by samvarankashyap** [comment]
centos-paas-test

**Comment by samvarankashyap** [comment]
centos-paas-test

File: cli/cli.py [hunk 1]
```diff
@@ -61,33 +61,33 @@ def find_layout(self, layout, lpf):
...
-    def lp_rise(self, lpf, target):
-        lpf = parse_yaml(lpf)
+    def lp_rise(self, pf, target):
+        pf = parse_yaml(pf)
         init_dir = os.getcwd()
         e_vars = {}
         e_vars['linchpin_config'] = self.get_config_path()
         e_vars['outputfolder_path'] = init_dir+"/outputs"
-        e_vars['inventory_outputs_path'] = init_dir+"/inventory"
+        e_vars['inventory_outputs_path'] = init_dir+"/inventories"
```
Comment (herlo) [inline cli/cli.py]: This needs to be a variable stored somewhere. I will address this in the linchpin_config. @samvarankashyap, does that sound reasonable to you?

File: cli/cli.py [hunk 2]
```diff
@@ -61,33 +61,33 @@ def find_layout(self, layout, lpf):
...
             print(e)
 
-    def lp_rise(self, lpf, target):
-        lpf = parse_yaml(lpf)
+    def lp_rise(self, pf, target):
+        pf = parse_yaml(pf)
         init_dir = os.getcwd()
         e_vars = {}
         e_vars['linchpin_config'] = self.get_config_path()
         e_vars['outputfolder_path'] = init_dir+"/outputs"
```
Comment (herlo) [inline cli/cli.py]: This needs to be a variable stored somewhere. I will address this in the linchpin_config. @samvarankashyap, does that sound reasonable to you?

Comment (samvarankashyap) [inline cli/cli.py]: @herlo : sure perfectly fine with storing the paths in config file.

**Comment by samvarankashyap** [comment]
The path variable referenced will stored in a common place . Most probably in linchpin_config file.

**Pull request merged**